baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/UnnecessarilyQualifiedTest.java

```diff
@@ -50,6 +50,31 @@ void testUnnecessarilyQualifiedType() {
                 .doTest();
     }
 
+    @Test
+    void testEnclosedClassNotFlaggedWhenImportIsPresent() {
+        fix().addInputLines(
+                        "Test.java",
+                        "import java.util.Map;",
+                        "import java.util.Map.Entry;",
+                        "import java.util.Set;",
+                        "class Test {",
+                        "  java.util.Map.Entry<?, ?> get() {",
+                        "    return null;",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import java.util.Map;",
+                        "import java.util.Map.Entry;",
+                        "import java.util.Set;",
+                        "class Test {",
+                        "  Map.Entry<?, ?> get() {",
```

Comment:
Some may prefer for this to end up as `Entry` rather than `Map.Entry`, but I'd argue that either one is better than `java.util.Map.Entry`.